src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ChunkOutputStream.java

```diff
@@ -447,15 +451,29 @@ public void showCallbackHtml(String htmlOutput)
          public void execute()
          {
             DomUtils.fillIFrame(frame.getIFrame(), htmlOutput);
+            DomUtils.forwardWheelEvent(frame.getIFrame().getContentDocument(), parentElement);
+            
+            Document anotherDoc = frame.getWindow().getDocument().getOwnerDocument();
             int contentHeight = frame.getWindow().getDocument().getDocumentElement().getOffsetHeight();
             frame.getElement().getStyle().setHeight(contentHeight, Unit.PX);
             frame.getElement().getStyle().setWidth(100, Unit.PCT);
             onHeightChanged();
 
             Command handler = () -> {
-               int newHeight = frame.getWindow().getDocument().getDocumentElement().getOffsetHeight();
-               frame.getElement().getStyle().setHeight(newHeight, Unit.PX);
-               onHeightChanged();
+               // reset height so we can shrink it if necessary
+               frame.getElement().getStyle().setHeight(0, Unit.PX);
+  
+               // delay calculating the height so any images can load
+               new Timer()
+               {
+                  @Override
+                  public void run()
+                  {
+                     int newHeight = frame.getWindow().getDocument().getDocumentElement().getOffsetHeight();
+                     frame.getElement().getStyle().setHeight(newHeight, Unit.PX);
+                     onHeightChanged();
+                  }
+               }.schedule(50);
```

Comment:
This could very well be happening over a WAN or slow connection so 50ms may not be enough time for everything to load (such that we are guaranteed to have the final height). Maybe we should wait for the frame's `load` event?

Reply:
Unfortunately the `load` event gets sent only to the iframe and isn't forwarded to the IDE. We have a `MutationObserver` monitoring the iframe for changes to know when to adjust the height, but the callback I've been using downloads images via HTTP requests so the observer recognizes that change and immediately adjusts the height, but before the image has downloaded. 
I talked with Jeff about this and it shouldn't be a major issue as the majority of the images they'll be sending will be small. We also discovered that if the `<img>` tag in the callback is passed with a `height` argument then we can circumvent the need to delay the load. I still like the idea of delaying it slightly if it's not noticeable. What do you think?

Reply:
Do we have any other way of detecting that the frame is fully loaded? I wonder if this should be a repeating command that basically polls until the iframe is "ready", and then runs only when ready.

Reply:
Unfortunately not that I'm aware of without requiring the user to include some logic in the callback (and having them set the img height is probably the simplest way); both `load` and `readystate` give false positives.